pkg/logger/logger.go

```diff
@@ -0,0 +1,132 @@
+
+package logger
+
+import (
+	"strings"
+	"sync"
+)
+
+const (
+	// LogTypeLog is normal log type
+	LogTypeLog = "log"
+	// LogTypeRequest is Request log type
+	LogTypeRequest = "request"
+
+	// Field names that defines Dapr log schema
+	logFieldTimeStamp = "time"
+	logFieldLevel     = "level"
+	logFieldType      = "log_type"
+	logFieldScope     = "scope"
+	logFieldMessage   = "message"
+	logFieldInstance  = "instance"
+	logFieldDaprVer   = "dapr_ver"
+	logFieldDaprID    = "dapr_id"
+)
+
+// LogLevel is Dapr Logger Level type
+type LogLevel string
+
+const (
+	// DebugLevel has verbose message
+	DebugLevel LogLevel = "debug"
+	// InfoLevel is default log level
+	InfoLevel LogLevel = "info"
+	// WarnLevel is for logging messages about possible issues
+	WarnLevel LogLevel = "warn"
+	// ErrorLevel is for logging errors
+	ErrorLevel LogLevel = "error"
+	// FatalLevel is for logging fatal messages. The system shuts down after logging the message.
+	FatalLevel LogLevel = "fatal"
+
+	// UndefinedLevel is for undefined log level
+	UndefinedLevel LogLevel = ""
+)
+
+// globalLoggers is the collection of Dapr Logger that is shared globally.
+// TODO: User will disable or enable logger on demand.
+var globalLoggers = map[string]Logger{}
+var globalLoggersLock = sync.RWMutex{}
+
+// Logger includes the logging api sets
+type Logger interface {
+	// EnableJSONOutput enables JSON formatted output log
+	EnableJSONOutput(enabled bool) error
+
+	// SetDaprID sets dapr_id field in log. Default value is empty string
+	SetDaprID(id string)
+	// SetOutputLevel sets log output level
+	SetOutputLevel(outputLevel LogLevel) error
+
+	// WithLogType specify the log_type field in log. Default value is LogTypeLog
+	WithLogType(logType string) Logger
+
+	// Info logs a message at level Info.
+	Info(args ...interface{})
+	// Infof logs a message at level Info.
+	Infof(format string, args ...interface{})
+	// Debug logs a message at level Debug.
+	Debug(args ...interface{})
+	// Debugf logs a message at level Debug.
+	Debugf(format string, args ...interface{})
+	// Warn logs a message at level Warn.
+	Warn(args ...interface{})
+	// Warnf logs a message at level Warn.
+	Warnf(format string, args ...interface{})
+	// Error logs a message at level Error.
+	Error(args ...interface{})
+	// Errorf logs a message at level Error.
+	Errorf(format string, args ...interface{})
+	// Fatal logs a message at level Fatal then the process will exit with status set to 1.
+	Fatal(args ...interface{})
+	// Fatalf logs a message at level Fatal then the process will exit with status set to 1.
+	Fatalf(format string, args ...interface{})
+}
+
+// toLogLevel converts to LogLevel
+func toLogLevel(level string) LogLevel {
+	switch strings.ToLower(level) {
+	case "debug":
+		return DebugLevel
+	case "info":
+		return InfoLevel
+	case "warn":
+		return WarnLevel
+	case "error":
+		return ErrorLevel
+	case "fatal":
+		return FatalLevel
+	}
+
+	// unsupported log level by Dapr
+	return UndefinedLevel
```

Comment:
Can we default to something?

Reply:
it is intended to check the validity of level